content/en-us/build/stateless/language-basic.md

```diff
@@ -0,0 +1,74 @@
+
+The prospering ecosystem is another reason to use Rust-based Ink! to develop your contract. There are many secure and high-performance libraries for you to use without building all the wheels from scratch.
+
+This section will not teach your about the Rust and Ink! language itself. Instead, it provides some helpful links and comments to get you prepared quickly.
+
+
+## Rust
+
+### Tooling
+
+When using Rust, you will always rely on the following official tools to manage your toolchain and compile your projects:
+- [rustup](https://rustup.rs/), which helps you install and manage different versions of Rust compilers;
+- cargo, the package manager installed automatically along with Rust compiler. Each Rust project contains a `Cargo.toml` file to describe its metadata and dependencies, and `cargo` downloads the dependencies and builds the project for you. The [cargo book](https://doc.rust-lang.org/cargo/reference/index.html) can help, and ensure you understand the [[dependencies]](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) and [[features]](https://doc.rust-lang.org/cargo/reference/features.html) sections in `Cargo.toml` before going on.
+
+### Language Basics
+
+The best material is surely the official [Rust book](https://doc.rust-lang.org/book/).
+- For a beginner, to learn the first 6 sections is enough for your journey in the Phat Contract world;
+- I also recommend to finish the 7 - 10 sections when you decide to go a little deeper.
+
+There are also many [books recommended by the community](https://www.reddit.com/r/rust/comments/sjclfb/best_book_to_learn_rust/) for more advanced topics.
+
+
+## Ink!
+
+Ink! is the programming language developed by Parity. The native Ink! contracts can run on different blockchains (like Polkadot and Phala), but Phala adds its secret ingredient called [pink-extension](https://github.com/Phala-Network/phala-blockchain/tree/master/crates/pink) (short for Phala Ink! Extension) to form Phat Contract which is only functional in Phala's off-chain Workers.
+
+> Phat Contract is the superset of Ink! contract.
+
+### Ink! Macros
+
+The difference between a Ink! contract and a normal Rust program is the macros in it. As shown in our [example contract](/en-us/build/getting-started/create-contract/#create-and-compile-your-first-contract), you need to annotate the Rust code with Ink! macros so the compiler can learn your contract structure. For example,
+
+```rust
+#[ink(storage)]
+pub struct OnchainState;
+```
+
+`#[ink(storage)]` tells us the following struct `OnchainState` saves the contract states and needs to be stored on-chain. And
+
+```rust
+#[ink(message)]
+pub fn query_handler(&self, from: AccountId) {
+    // actual implementation
+}
+
+#[ink(message)]
+pub fn transaction_handler(&mut self, from: AccountId) {
```

Comment:
Transactions are for stateful program. Better to not mention it until the next section

Reply:
removed